dbos/_utils.py

```diff
@@ -15,3 +18,31 @@ class GlobalParams:
     except importlib.metadata.PackageNotFoundError:
         # If package is not installed or during development
         dbos_version = "unknown"
+
+
+def retriable_postgres_exception(e: DBAPIError) -> bool:
+    if isinstance(e.orig, psycopg.OperationalError):
+        driver_error: psycopg.OperationalError = e.orig
+        pgcode = driver_error.sqlstate or ""
+        # Failure to establish connection
+        if "connection failed" in str(driver_error):
+            return True
+        # Error within database transaction
+        elif "server closed the connection unexpectedly" in str(driver_error):
+            return True
+        # Connection timeout
+        if isinstance(driver_error, psycopg.errors.ConnectionTimeout):
+            return True
+        # Insufficient resources
+        elif pgcode.startswith("53"):
+            return True
+        # Connection exception
+        elif pgcode.startswith("08"):
+            return True
+        # Operator intervention
+        elif pgcode.startswith("57"):
+            return True
+        else:
+            return False
+    else:
+        return False
```

Comment:
This is a bit worrying -- how could we get a more definitive evaluation?

Reply:
This is the best psycopg gives us, from my time reading through their code